Common/cpp/SharedItems/Shareables.h

```diff
@@ -58,33 +81,40 @@ class Shareable {
 template <typename BaseClass>
 class RetainingShareable : virtual public BaseClass {
  private:
-  std::shared_ptr<JSRuntimeHelper> runtimeHelper_;
-  std::unique_ptr<jsi::Value> remoteValue_;
+  jsi::Runtime *rnRuntime_;
+  jsi::Runtime *uiRuntime_;
```

Comment:
If I understand the below code correctly, this is meant to be used with other non-ui runtime. Maybe we can change the naming and make it be "primaryRuntime" and "secondaryRuntime instead rnRuntime and uiRuntime. Also, we should add some comments here to explain that this class is optimized for passing shareables between two runtimes (typically it'd be main RN runtime and some secondary runtime, e.g. UI one)

Reply:
That's true, both `RetainingShareable` and `ShareableSynchronizedDataHolder` are supposed to support arbitrary runtimes. I've renamed the variables according to your suggestion and `secondaryValue_` looks quite bad. While I like the idea of using generalized names here , `rn` and `ui` prefixes make the code easier to understand and reason about, especially considering that this is the only use case as for now.